chain/chunks/src/lib.rs

```diff
@@ -690,6 +706,17 @@ impl ShardsManager {
                     shard_id,
                     &chunk_hash,
                     false,
+                    // Request from a single node for missing chunks in old blocks
+                    // This is to save the number of requests a node sends
+                    // For old blocks, we can be reasonably sure that some peer has downloaded
+                    // the full chunk and we can get the chunk from them
+                    // TODO: when we enable challenges, this may break data availability
+                    //    if not many nodes receive the chunk before a block gets old (no longer the head)
+                    //    We leave this issue here for now because it is not a problem now when
+                    //    each shard is tracked by the majority of nodes in the network
+                    //    and we may change the way how chunks are distributed to non-validators
+                    //    https://github.com/near/nearcore/issues/3890
+                    old_block,
                     old_block,
```

Comment:
we have a lot of parameters here -- as Rust doesn't support named params, consider adding an inline comment to make reading easier:

false,     // force_request_full
old_block, // request_from_single_source 

etc..